docs/design/proposals/rbd-volume-healer.md

```diff
@@ -0,0 +1,94 @@
+# RBD NBD VOLUME HEALER
+
+- [RBD NBD VOLUME HEALER](#rbd-nbd-volume-healer)
+  - [Rbd Nbd](#rbd-nbd)
+    - [Advantages of userspace mounters](#advantages-of-userspace-mounters)
+    - [Side effects of userspace mounters](#side-effects-of-userspace-mounters)
+  - [Volume Healer](#volume-healer)
+    - [More thoughts](#more-thoughts)
+
+## Rbd Nbd
+
+The RBD CSI plugin will provision new RBD images and attach and mount those
+to workloads. Currently, the default mounter is krbd, which uses the kernel
+rbd driver to mount the rbd images onto the application node. Here on
+at cephcsi we will also have a userspace way of mounting the RBD images,
+via RBD-NBD.
+
+[Rbd-nbd](https://docs.ceph.com/en/latest/man/8/rbd-nbd/) is a client for
+RADOS block device (rbd) images like the existing rbd kernel module. It
+will map an rbd image to an NBD (Network Block Device) device, allowing
+access to it as a regular local block device.
+
+![csi-rbdnbd](https://user-images.githubusercontent.com/12432241/125400940-23e75d00-e3d0-11eb-806c-bab09318de54.png)
+
+It’s worth making a note that the rbd-nbd processes will run on the
+client-side, which is inside the `csi-rbdplugin` node plugin.
+
+### Advantages of userspace mounters
+
+- Adding changes to client logic to accommodate new features will be easy
```

Comment:
I assume that this is out of scope for ceph-csi, make it clear that it is for the Ceph project, compare to krbd?

Reply:
users who don't have much understanding of the ceph or storage, might not understand the advantages of userspace mounter. Does it hurt to capture these details as an intro to `what and why userspace mounter`?

Reply:
it surely does not hurt, but currently the text is not clear where it becomes easier to accommodate new features. You mean that it is easier to add features to rbd-nbd as that is released regularly with Ceph, and more difficult to add features to the kernel rbd module as that is part of the Linux kernel release schedule? (I understand it is about adding features to rbd-nbd here, and not ceph-csi, but that might be clarified a little too.)